[Dropbox] Documentation for Advanced sync rules (#1322)
elastic/connectors-python: docs/sync-rules/DROPBOX.md

> ### Setting up the Jira connector

Reviewer: ```suggestion
### Setting up the Dropbox connector
```

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

### Example advanced sync rules

> #### Two rules for indexing content based on the query only

Reviewer: ```suggestion
#### Two rules for indexing content based on queries only
```

```json
[
  {
    "query": "confidential"
  },
  {
    "query": "dropbox"
  }
]
```

> #### One query for indexing data based on the file_extensions

Reviewer: ```suggestion
#### Single query for indexing data based on file extensions
```

```json
[
  {
    "query": "dropbox",
    "options": {
      "file_extensions": [
        "txt",
        "pdf"
      ]
    }
  }
]
```

> #### One query for indexing data based on file_categories

Reviewer: ```suggestion
#### Single query for indexing data based on file categories
```

```json
[
  {
    "query": "test",
    "options": {
      "file_categories": [
        {
          ".tag": "paper"
        },
        {
          ".tag": "png"
        }
      ]
    }
  }
]
```
**Note** 

- `query` contains a string that matches words in the filename.

> - In case, both `file_extensions` and `file_categories` are provided, priority is given to `file_categories`.

Reviewer: ```suggestion
- If both `file_extensions` and `file_categories` are provided, priority is given to `file_categories`.
```

### Limitations

> - Content extraction is not supported for paper files with advanced sync rules enabled.

Reviewer: ```suggestion
- Content extraction is not supported for Dropbox Paper files when advanced sync rules are enabled.
```
